pkg/handler/snapshots.go

```diff
@@ -327,6 +330,51 @@ func (sh *SnapshotHandler) bulkDeleteSnapshot(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+// GetTemplateRepoConfigurationFiles godoc
+// @Summary      Get configuration files for all repositories in a template
+// @ID           getTemplateRepoConfigurationFiles
+// @Tags         repositories
+// @Accept       json
+// @Produce      text/plain
+// @Param        template_uuid  path  string    true  "Identifier of the template"
+// @Success      200   {string} string
+// @Failure      400 {object} ce.ErrorResponse
+// @Failure      401 {object} ce.ErrorResponse
+// @Failure      404 {object} ce.ErrorResponse
+// @Failure      500 {object} ce.ErrorResponse
+// @Router       /templates/{template_uuid}/config.repo [get]
+func (sh *SnapshotHandler) getTemplateRepoConfigurationFiles(c echo.Context) error {
+	_, orgID := getAccountIdOrgId(c)
+	templateUUID := c.Param("template_uuid")
+
+	xRHID := identity.GetIdentity(c.Request().Context())
+	if xRHID.Identity.Type == "System" {
+		if orgID != xRHID.Identity.OrgID && orgID != xRHID.Identity.Internal.OrgID {
+			return ce.NewErrorResponse(http.StatusForbidden, "Error authorizing client", "System OrgID does not match template OrgID")
+		}
+	}
+
+	template, err := sh.DaoRegistry.Template.Fetch(c.Request().Context(), orgID, templateUUID, false)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			err = &ce.DaoError{NotFound: true, Message: "Could not find template with UUID " + templateUUID}
+		}
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching template", err.Error())
+	}
+
+	var configFiles strings.Builder
+	for _, snap := range template.Snapshots {
+		templateRepoConfigFile, err := sh.DaoRegistry.Snapshot.GetRepositoryConfigurationFile(c.Request().Context(), orgID, snap.UUID, false, true, templateUUID)
```

Comment:
I haven't fully thought out this implementation so feel free to disagree with me here

We may want to create a TemplateDao method like `GetRepoConfigFiles(ctx, org, uuid)`, which returns the whole "configFiles" string. Under the hood I think it could still call the `Snapshot.GetRepoConfigFile` method, but it could mean we don't have parameters that in some cases go unused.